acloud/provider.go

```diff
@@ -0,0 +1,69 @@
+package acloud
+
+import (
+	"fmt"
+	"goscraper/local"
+	"goscraper/proxy"
+)
+
+type ACloudProvider struct {
+	ACloudEnv *local.ACloudEnv
+	proxy.Connection
+	SandboxCredentials
+}
+
+func (p *ACloudProvider) Login(username, password string) (err error) {
+
+	//load env credentials from .env file
+	login, err := local.LoadEnv()
+	local.PanicIfErr(err)
+	fmt.Println("login : ", login)
+
+	//connect to website
+	connect, err := proxy.Login(proxy.WebsiteLogin{Url: login.Url, Username: username, Password: password})
```

Comment:
Can you explain what the thinking behind adding this `Login` method to the `proxy` package and calling it from here? This function is supposed to be the one implementing the interface method, it seems like an anti-pattern to then embed your business logic for it back in the package that provided the interface.

Reply:
Yes, it is an antipattern/confusing that I left that code in the /proxy folder with the original provider interface, and put the implementation methods of the interface in provider.go in the /acloud folder.  What I was hoping to do was make the methods in carrierproxy.go as generalizable as possible, so that other folders could be added with integration-specific code (what I intended the /acloud folder to be). I will do some reorganizing.